src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -5386,11 +5392,8 @@ def _subroutine_handler(self, node, parent):
                 # declared with its own local name. Be wary that this
                 # function may not be referenced so there might not be
                 # a RoutineSymbol.
-                try:
-                    routine_symbol = routine.symbol_table.lookup(routine.name)
-                    routine_symbol.datatype = base_type
-                except KeyError:
-                    pass
```

Comment:
Yes, I think the KeyError cannot happen any more because the Routine will always have a container which can contain the symbol - previously this would fail on `FileContainer`s but I don't think it will any more.

Reply:
Then you should remove the second part of the comment :)

(Anyway your other PR will maybe get rid of this altogether)